src/lib/swapper/swappers/ThorchainSwapper/utils/getLimit/getLimit.ts

```diff
@@ -46,21 +46,17 @@ export const getLimit = async ({
     deps,
   })
 
-  // TODO(gomes): grep for maybeTradeRate.match() and bubble the error up
-  // For now, we're just doing the same flow as before and returning either an actual rate or a minimum
-  const tradeRate = await maybeTradeRate.match({
-    ok: rate => Promise.resolve(rate),
+  const tradeRateBelowMinimum = await getTradeRateBelowMinimum({
+    sellAssetId: sellAsset.assetId,
+    buyAssetId,
+    deps,
+  })
+  const tradeRate = maybeTradeRate.unwrapOr(
     // TODO: Handle TRADE_BELOW_MINIMUM specifically and return a result here as well
     // Though realistically, TRADE_BELOW_MINIMUM is the only one we should really be seeing here,
     // safety never hurts
-    err: _err =>
-      getTradeRateBelowMinimum({
-        sellAssetId: sellAsset.assetId,
-        buyAssetId,
-        deps,
-      }),
-  })
-
+    tradeRateBelowMinimum.unwrap(),
```

Comment:
Do we know that `tradeRateBelowMinimum` has a result here?

Reply:
In this particular case, it realistically should, though it may not: 9399ade0a71cf571d105c2644d0e5238d5d379f7